Enable execution of any arch on TC.

mari-crl opened the pull request:
To build amd64 Docker images, one must run amd64 containers.
To build arm64 Docker images, one must run arm64 containers.
To run a container, one must be able to execute its binfmt.
To execute a non-native binfmt, emulation must be enabled.

This change enables emulation for our TC agents so they can run binaries
of any architecture. This allows them to build multi-arch images on
any of the architectures involved in the result by emulating the others.

Part of CRDB-3143

Release note: None.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/90152)
<!-- Reviewable:end -->

rail reviewed (approved):
LGTM, :shipit:

mari-crl committed:
Enable execution of any arch on TC.

To build amd64 Docker images, one must run amd64 containers.
To build arm64 Docker images, one must run arm64 containers.
To run a container, one must be able to execute its binfmt.
To execute a non-native binfmt, emulation must be enabled.

This change enables emulation for our TC agents so they can run binaries
of any architecture. This allows them to build multi-arch images on
any of the architectures involved in the result by emulating the others.

Part of CRDB-3143

Release note: None.

rail reviewed (approved):
LGTM, :shipit:

mari-crl commented:
Let's make it so!

bors r+
